src/Psalm/Internal/Provider/ParserCacheProvider.php

```diff
@@ -402,4 +400,18 @@ private function getParserCacheKey(string $file_name): string
     {
         return md5($file_name) . ($this->use_igbinary ? '-igbinary' : '') . '-r';
     }
+
+    private function createCacheDirectory(string $parser_cache_directory): void
+    {
+        if (!is_dir($parser_cache_directory)) {
```

Comment:
wouldn't `@mkdir(...)` make more sense? After all, we only want the directory to exist. On the other hand this could just move race-condition elsewhere.

Reply:
_Technically_ it could be a file for some reason, in which case an error should be thrown, but I wouldn't really be opposed to changing it to `@mkdir(...)`, as it being a file seems incredibly unlikely.

I suppose you're right that this could just move the race condition somewhere else, I hadn't considered that. I haven't really looked into how the caching works, hopefully different processes aren't going to be using the same `$file_cache_key`.

Reply:
`mkdir` could also fail due to lack of write permission, which also seems unlikely, but more likely than the directory path being a file.